### PR TITLE
Introduce Quiltflower decompiler support

### PR DESCRIPTION
**Pull request by geoand**
[Quiltflower](https://github.com/QuiltMC/quiltflower) is a new decompiler based on Fernflower
that generates better decompiled code.

This change also deprecates the Fernflower configuration
in favor of the new Quiltflower configuration

### CODE REVIEW

**Comment by gsmet** [comment]
I saw this one the other day. What I was wondering is if it wasn't better to keep the one maintained by Jetbrains, for which we are sure it's going to be around for a while?
It's an open question, I don't have a strong opinion and Jetbrains might decide to switch to another one themselves.

**Comment by geoand** [comment]
This seems to be more actively maintained

**Comment by geoand** [comment]
Moreover, this one has actual Github releases as opposed to Fernflower which only has some hacky support for some of the commits made in the initial repo

**Review by gsmet — CHANGES_REQUESTED** [review]
Fair enough. I suppose we will see how it goes, it's an advanced feature anyway.

This doc needs an update: https://quarkus.io/guides/writing-extensions .

**Comment by gsmet** [comment]
Should have been more precise: search for `fernflower` in it.

**Comment by geoand** [comment]
> Fair enough. I suppose we will see how it goes, it's an advanced feature anyway.
> 
> This doc needs an update: https://quarkus.io/guides/writing-extensions .

done

File: docs/src/main/asciidoc/writing-extensions.adoc [hunk 1]
```diff
@@ -2164,7 +2164,7 @@ The only particular aspect of writing Quarkus extensions in Eclipse is that APT
 Quarkus generates a lot of classes during the build phase and in many cases also transforms existing classes.
 It is often extremely useful to see the generated bytecode and transformed classes during the development of an extension.
 
-If you set the `quarkus.package.fernflower.enabled` property to `true` then Quarkus will download and invoke the https://github.com/JetBrains/intellij-community/tree/master/plugins/java-decompiler/engine[Fernflower decompiler] and dump the result in the `decompiled` directory of the build tool output (`target/decompiled` for Maven for example).
+If you set the `quarkus.package.quiltflower.enabled` property to `true` then Quarkus will download and invoke the https://github.com/JetBrains/intellij-community/tree/master/plugins/java-decompiler/engine[Fernflower decompiler] and dump the result in the `decompiled` directory of the build tool output (`target/decompiled` for Maven for example).
```
Comment (gsmet) [inline docs/src/main/asciidoc/writing-extensions.adoc:2167]: You should also update the link and the link title a bit further in the sentence. Sorry about the iterations.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/writing-extensions.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/writing-extensions.adoc:2167]: Good point, sorry for missing it when updating the docs

**Review by gsmet — APPROVED** [review]
Nice! Let's wait for CI.

**Comment by Asbestosstar** [comment]
Awesome.

**Pull request merged**